### PR TITLE
test: make test TestTopSQLCPUProfile stable by mock high cpu usage

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve?

```sql
[2021-06-23T05:48:54.266Z] ----------------------------------------------------------------------
[2021-06-23T05:48:54.266Z] FAIL: tidb_test.go:1196: tidbTestTopSQLSuite.TestTopSQLCPUProfile
[2021-06-23T05:48:54.266Z] 
[2021-06-23T05:48:54.266Z] tidb_test.go:1385:
[2021-06-23T05:48:54.266Z]     checkFn(ca.sql, ca.planRegexp)
[2021-06-23T05:48:54.266Z] tidb_test.go:1365:
[2021-06-23T05:48:54.266Z]     // since 1 sql may has many plan, check `len(stats) > 0` instead of `len(stats) == 1`.
[2021-06-23T05:48:54.266Z]     c.Assert(len(stats) > 0, IsTrue, commentf)
[2021-06-23T05:48:54.266Z] ... obtained bool = false
[2021-06-23T05:48:54.266Z] ... sql: select * from t where a in (1,2,3,4
```

### What is changed and how it works?

Some SQL is run so fast that pprof profile unable to sample those SQL, this PR use failpoint to inject some cpu load for each SQL to make sure each SQL can be sampled by Top SQL.

### Related changes

- N/A

### Check List 

Tests 

- No code

Side effects

- N/A

### Release note 

- No release note.

### CODE REVIEW

File: server/tidb_test.go [hunk 1]
```diff
@@ -1233,8 +1236,8 @@ func (ts *tidbTestTopSQLSuite) TestTopSQLCPUProfile(c *C) {
 		{sql: "update t set b=a where b is null limit 1;", planRegexp: ".*Limit.*TableReader.*"},
 		{sql: "delete from t where b = a limit 2;", planRegexp: ".*Limit.*TableReader.*"},
 		{sql: "replace into t (b) values (1),(1),(1),(1),(1),(1),(1),(1);", planRegexp: ""},
-		{sql: "select * from t use index(idx) where a<10000000;", planRegexp: ".*IndexLookUp.*"},
-		{sql: "select * from t ignore index(idx) where a>0;", planRegexp: ".*TableReader.*"},
+		{sql: "select * from t use index(idx) where a<10;", planRegexp: ".*IndexLookUp.*"},
+		{sql: "select * from t ignore index(idx) where a>1000000000;", planRegexp: ".*TableReader.*"},
```
Comment (crazycs520) [inline server/tidb_test.go:1240]: Change those were try to avoid read/write conflict.

**Comment by tisonkun** [comment]
ref #25122

**Review by zimulala — APPROVED** [review]
LGTM

File: util/topsql/topsql.go [hunk 2]
```diff
@@ -66,6 +71,37 @@ func AttachSQLInfo(ctx context.Context, normalizedSQL string, sqlDigest *parser.
...
+			if !isDML {
+				failpoint.Return(ctx)
+			}
+			start := time.Now()
+			logutil.BgLogger().Info("attach SQL info", zap.String("sql", normalizedSQL), zap.Bool("has-plan", len(normalizedPlan) > 0))
+			for {
+				if time.Since(start) > 11*time.Millisecond {
+					break
+				}
+				for i := 0; i < 10e5; i++ {
```
Comment (AilinKid) [inline util/topsql/topsql.go:100]: what's is loop this for ?

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Review by tangenta — APPROVED** [review]

File: util/topsql/topsql.go (hunk 2 above)
Comment (crazycs520) [reply util/topsql/topsql.go:100]: Just use for consume CPU time.

**Review by AilinKid — APPROVED** [review]

**Pull request merged**